recursion/program/src/reduce.rs

```diff
@@ -208,7 +260,27 @@ pub fn build_reduce_program(setup: bool) -> RecursionProgram<Val> {
             },
             // Recursive proof
             |builder| {
-                // TODO: Verify proof public values
+                let mut pv_elements = Vec::new();
+                for i in 0..PROOF_MAX_NUM_PVS {
+                    let element = builder.get(&proof.public_values, i);
+                    pv_elements.push(element);
+                }
+
+                let proof_pv = RecursionPublicValues::<Felt<_>>::from_vec(pv_elements);
```

Comment:
are we planning to remove the hashing part and just commit to all the public values directly?

Reply:
I plan to still keep the hashing part, but the individual public values will be more structured instead of an array.